cmd/buildah/addcopy.go

```diff
@@ -20,63 +21,56 @@ type addCopyResults struct {
 	contextdir string
 }
 
+func createCommand(addCopy string, desc string, short string, opts *addCopyResults) *cobra.Command {
+	return &cobra.Command{
+		Use:   addCopy,
+		Short: short,
+		Long:  desc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return addAndCopyCmd(cmd, args, strings.ToUpper(addCopy), *opts)
+		},
+		Example: `buildah ` + addCopy + ` containerID '/myapp/app.conf'
+  buildah ` + addCopy + ` containerID '/myapp/app.conf' '/myapp/app.conf'`,
+		Args: cobra.MinimumNArgs(1),
+	}
+}
+
+func applyFlagVars(flags *pflag.FlagSet, opts *addCopyResults) {
+	flags.SetInterspersed(false)
+	flags.BoolVar(&opts.addHistory, "add-history", false, "add an entry for this operation to the image's history.  Use BUILDAH_HISTORY environment variable to override. (default false)")
+	flags.StringVar(&opts.chown, "chown", "", "set the user and group ownership of the destination content")
+	flags.StringVar(&opts.chmod, "chmod", "", "set the access permissions of the destination content")
+	flags.StringVar(&opts.ignoreFile, "ignorefile", "", "path to .dockerignore file")
+	flags.StringVar(&opts.contextdir, "contextdir", "", "context directory path")
+	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "don't output a digest of the newly-added/copied content")
+}
+
 func init() {
 	var (
 		addDescription  = "\n  Adds the contents of a file, URL, or directory to a container's working\n  directory.  If a local file appears to be an archive, its contents are\n  extracted and added instead of the archive file itself."
 		copyDescription = "\n  Copies the contents of a file, URL, or directory into a container's working\n  directory."
+		shortAdd        = "Add content to the container"
+		shortCopy       = "Copy content into the container"
 		addOpts         addCopyResults
 		copyOpts        addCopyResults
 	)
-	addCommand := &cobra.Command{
-		Use:   "add",
-		Short: "Add content to the container",
-		Long:  addDescription,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return addCmd(cmd, args, addOpts)
-		},
-		Example: `buildah add containerID '/myapp/app.conf'
-  buildah add containerID '/myapp/app.conf' '/myapp/app.conf'`,
-		Args: cobra.MinimumNArgs(1),
-	}
+	addCommand := createCommand("add", addDescription, shortAdd, &addOpts)
```

Comment:
Suggestion only, I think it would be more clear if you did:
```suggestion
	addCommand := createCommand("ADD", addDescription, shortAdd, &addOpts)
```
It would stand out for me more when I searched for "ADD" from line 124.   Ditto uppercasing "copy" to "COPY" at line 60.  I'd still keep the ToUpper() call in play above in case someone forgets to uppercase stuff in further development.

Reply:
I have to protest this one :) It will force 3x toLower in the createCommand which is less efficient than 1x toUpper.

Reply:
Line 26, 32 and 33 use lower

Reply:
Yes that is what I mean

Reply:
Lint is not happy. They are starting from a capital letter and they should have a comment as they will be treated as exported.
I can do this, but honestly don't see a point.
I would recommend keeping current solution.
Let me know what you think.

Reply:
I can live with it as is.  However, as I was reviewing, I saw your snazzy check for "ADD" and couldn't find "ADD" being passed anywhere, it took me 10 minutes or so (again late night brain) to figure it out.  The suggestion was to try to make the code a bit clearer, but not completely necessary, especially given the lint fun you've since run into.

Thanks for the consideration though.

Reply:
Thanks for the comment. I personally think this check is sneaky, not snazzy, but in my opinion acceptable. Especially that it is a minor side effect of code unification.